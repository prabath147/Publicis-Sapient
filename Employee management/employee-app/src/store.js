import { createStore, applyMiddleware } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import rootReducer from "./Component/reducers";
import Thunk from "redux-thunk";

const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(Thunk))
);

export default store;
