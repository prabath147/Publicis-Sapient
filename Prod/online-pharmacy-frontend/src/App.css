
html,body {
  margin: 0;
  padding: 0;
  height: 100%;
  overflow-x: hidden;
}

.main-nav {
  margin-top: 2.5rem;
  width: 6rem;
}


.wel-msg{
  font-size: 1.8rem;
  font-family: 'Poppins', sans-serif;
}

.register{
  font-size: 1.8rem;
  font-family: 'Poppins', sans-serif;
}

