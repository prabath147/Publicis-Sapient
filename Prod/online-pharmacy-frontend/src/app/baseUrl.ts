export const baseURL = "https://frk5iphcfk.execute-api.us-east-1.amazonaws.com/";
    // "http://localhost:8081";
    
