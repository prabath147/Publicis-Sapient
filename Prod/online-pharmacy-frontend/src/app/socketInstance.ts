
export const socketURL = "https://c8zpvs7fv4.execute-api.us-east-1.amazonaws.com/notify-ws"

// const socket = new SockJS(socketURL);
// export const stompClient = Stomp.over(socket);