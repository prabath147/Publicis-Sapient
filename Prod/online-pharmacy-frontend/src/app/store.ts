import { Action, configureStore, ThunkAction } from "@reduxjs/toolkit";
import StoreReducer from "../components/pages/Admin/storeList/StoreSlice";
import SubscriptionReducer from "../components/pages/AdminSubscriptionPack/SubscriptionvSlice";
import UserReducer from "../components/pages/Auth/Login/UserSlice";
import CartReducer from "../components/pages/Cart/CartSlice";
import ManagerReducer from "../components/pages/Manager/ManagerSlice";
import NotificationReducer from "../components/pages/Open/notificationList/NotificationSlice";
// import OrderSlice from "../components/pages/orderDetail/OrderSlice";
// import OrderListSlice from "../components/pages/orderList/OrderListSlice";

export const store = configureStore({
  reducer: {
    storeL: StoreReducer,
    user: UserReducer,
    notification: NotificationReducer,
    viewsubscription: SubscriptionReducer,
    // addsubscription: AddSubcriptionRe,
    cart: CartReducer,
    // order: OrderSlice,
    // orderL: OrderListSlice,
    manager: ManagerReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
