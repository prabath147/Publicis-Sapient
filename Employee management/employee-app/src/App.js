// import logo from './logo.svg';
import './App.css';
import EmployeeInfo from './Component/pages/EmployeeInfo';

import EmployeeRecords from './Component/pages/EmployeeRecords';
import Leave from './Component/pages/Leave';
import LeaveApproval from './Component/pages/LeaveApproval';
import Login from './Component/pages/Login';
import SignUp from './Component/pages/SignUp';
import { BrowserRouter, Route, Routes } from "react-router-dom";
import { Provider } from "react-redux";
import store from "./store";
import EditLeave from './Component/pages/EditLeave';
import LeaveStatus from './Component/pages/LeaveStatus';
import EditEmployee from './Component/pages/EditEmployee';
import Home from './Component/pages/Home';


function App() {
  return (
    <div className="App">
      <Provider store={store}>
      {/* <EmployeeRecord1/> */}
      <BrowserRouter>
        
        <br />
        <div className="container">
          <Routes>
          <Route path="/" element={<Login />} />
            <Route path="/login" element={<Login />} />
            <Route path="/signup" element={<SignUp/>}/>
            <Route path="/home" element={<Home />} />
            <Route path="/logout" element={<Login />} />
            <Route path="/employee/records" element={<EmployeeRecords/>}/>
              <Route path="/employee/info" element={<EmployeeInfo />} />
              <Route path="/leave" element={<Leave />} />
              <Route path="/leave/edit/:id" element={<EditLeave/>} />
            <Route path="/leave/approval" element={<LeaveApproval />} />
            <Route path="/leave/status" element={<LeaveStatus />} />
            <Route path="/employee/edit" element={<EditEmployee />} />
            
          </Routes>
        </div>
      </BrowserRouter>
      </Provider>
    </div>
  );
}

export default App;
