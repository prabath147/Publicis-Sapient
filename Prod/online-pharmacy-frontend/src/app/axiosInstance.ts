import axios, { AxiosRequestConfig } from "axios";
import LocalUserService from "../services/LocalUserService";
import TokenService from "../services/TokenService";
import { baseURL } from "./baseUrl";

const axiosInstance = axios.create({
  baseURL: baseURL,
  timeout: 5000,
  headers: {
    "Content-Type": "application/json",
    accept: "application/json",
  },
});

axiosInstance.interceptors.request.use(
  (config: AxiosRequestConfig) => {
    const token = TokenService.getLocalAccessToken();
    if (token && config.headers !== undefined) {
      config.headers["Authorization"] = "Bearer " + token; // for Spring Boot back-end
      // config.headers["x-access-token"] = token; // for Node.js Express back-end
      config.headers["ngrok-skip-browser-warning"] = "";
    }
    return config;
  },
  (error) => {
    return Promise.reject(error);
  }
);

axiosInstance.interceptors.response.use(
  (res) => {
    return res;
  },
  async (err) => {
    const originalConfig = err.config;
    if (originalConfig.url !== "api/auth/signin" && err.response) {
      // Access Token was expired
      if (err.response.status === 401 && !originalConfig._retry) {
        originalConfig._retry = true;
        try {
          const rs = await axiosInstance.post("api/auth/refresh-token", {
            refreshToken: TokenService.getLocalRefreshToken(),
          });
          const { accessToken } = rs.data;
          TokenService.updateLocalAccessToken(accessToken);
          return axiosInstance(originalConfig);
        } catch (_error) {
          TokenService.removeLocalAccessToken()
          TokenService.removeLocalRefreshToken()
          LocalUserService.removeLocalUser()
          // window.location.reload();
          return Promise.reject(_error);
        }
      }
    }
    return Promise.reject(err);
  }
);
export default axiosInstance;
