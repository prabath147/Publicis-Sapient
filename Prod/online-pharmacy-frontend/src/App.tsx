import { showNotification } from "@mantine/notifications";
import { useEffect } from "react";
import { BrowserRouter } from "react-router-dom";
import { useAppDispatch, useAppSelector } from "./app/hooks";
import {
  getUserData,
  isCustomer,
  LoginAction
} from "./components/pages/Auth/Login/UserSlice";
import { getCartAPI } from "./components/pages/Cart/CartAPI";
import { loadCart } from "./components/pages/Cart/CartSlice";
import AppRouter from "./components/ui/appRouter/AppRouter";
import LocalUserService from "./services/LocalUserService";
import TokenService from "./services/TokenService";

function App() {
  const dispatch = useAppDispatch();

  const refresh_token = TokenService.getLocalRefreshToken();
  const token = TokenService.getLocalAccessToken();
  const user = LocalUserService.getLocalUser();
  const cuser = useAppSelector(getUserData);
  const isCustomerType = useAppSelector(isCustomer)


  if (refresh_token !== "") {
    dispatch({
      type: LoginAction,
      payload: {
        token: token,
        refreshToken: refresh_token,
        type: "Bearer",
        id: user.id,
        email: user.email,
        username: user.username,
        roles: [""]
      },
    });
  }
  useEffect(() => {
    if (cuser.id !== -1 && isCustomerType) {
      getCartAPI(cuser.id)
        .then((response) => {
          // console.log(response.data.items);
          dispatch(loadCart(response.data.items))
        })
        .catch((error) => {
          showNotification({ color: 'red', message: "oops, something went wrong" })
          console.log(error);
        });
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [cuser.id]);



  return (
    <div data-testid="app">
      <BrowserRouter>
        <AppRouter />
      </BrowserRouter>
    </div>
  );
}

export default App;
