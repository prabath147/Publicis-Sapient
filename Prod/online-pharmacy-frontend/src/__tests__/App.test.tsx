import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store } from '../app/store';
import App from '../App';

test('renders learn react link', () => {
  const component = renderer.create(
    <Provider store={store}>
      <App />
    </Provider>,
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
}
);
