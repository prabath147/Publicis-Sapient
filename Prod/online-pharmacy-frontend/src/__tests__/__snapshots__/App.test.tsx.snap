// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div
  data-testid="app"
>
  <div
    className="mantine-Container-root mantine-15pjuqq"
  >
    <div
      className="main-nav"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <div
          className="mantine-Image-root mantine-15po0m8"
          style={
            Object {
              "height": "auto",
              "width": "100%",
            }
          }
        >
          <figure
            className="mantine-11nhzn5 mantine-Image-figure"
          >
            <div
              className="mantine-qqmv3w mantine-Image-imageWrapper"
            >
              <img
                className="mantine-3y8yz3 mantine-Image-image"
                onError={[Function]}
                onLoad={[Function]}
                src="logo.jpg"
                style={
                  Object {
                    "height": "auto",
                    "objectFit": "cover",
                    "width": "100%",
                  }
                }
              />
            </div>
          </figure>
        </div>
      </a>
    </div>
  </div>
  <div>
    <div
      className="mantine-Container-root mantine-15pjuqq"
    >
      <div
        className="mantine-rl3tvd"
      >
        <div
          className="mantine-vdkfn5"
        >
          <h1
            className="mantine-Text-root mantine-Title-root mantine-10qiqy4"
          >
            <span
              className="mantine-p6tf2k"
            >
              PillZone
            </span>
            - An Online Marketplace for Pharmaceutical Products!
          </h1>
          <div
            className="mantine-Text-root mantine-qw4l90"
          >
            We offer a huge catalog of medicines and health care products, sourced from various offline pharmacies and medicine companies.
          </div>
          <ul
            className="mantine-List-root mantine-ephyri"
          >
            <li
              className="mantine-List-item mantine-List-withIcon mantine-990sqx"
            >
              <div
                className="__mantine-ref-itemWrapper mantine-1hf9p21 mantine-List-itemWrapper"
              >
                <span
                  className="mantine-amxt3q mantine-List-itemIcon"
                >
                  <div
                    className="mantine-ThemeIcon-root mantine-17dkyrq"
                  >
                    <svg
                      className="icon icon-tabler icon-tabler-check"
                      fill="none"
                      height={12}
                      stroke="currentColor"
                      strokeLinecap="round"
                      strokeLinejoin="round"
                      strokeWidth={1.5}
                      viewBox="0 0 24 24"
                      width={12}
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                        stroke="none"
                      />
                      <path
                        d="M5 12l5 5l10 -10"
                      />
                    </svg>
                  </div>
                </span>
                <span>
                  <b>
                    Tactically Optimized Inventory
                  </b>
                </span>
              </div>
            </li>
            <li
              className="mantine-List-item mantine-List-withIcon mantine-990sqx"
            >
              <div
                className="__mantine-ref-itemWrapper mantine-1hf9p21 mantine-List-itemWrapper"
              >
                <span
                  className="mantine-amxt3q mantine-List-itemIcon"
                >
                  <div
                    className="mantine-ThemeIcon-root mantine-17dkyrq"
                  >
                    <svg
                      className="icon icon-tabler icon-tabler-check"
                      fill="none"
                      height={12}
                      stroke="currentColor"
                      strokeLinecap="round"
                      strokeLinejoin="round"
                      strokeWidth={1.5}
                      viewBox="0 0 24 24"
                      width={12}
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                        stroke="none"
                      />
                      <path
                        d="M5 12l5 5l10 -10"
                      />
                    </svg>
                  </div>
                </span>
                <span>
                  <b>
                    Least Price Centered Design
                  </b>
                </span>
              </div>
            </li>
            <li
              className="mantine-List-item mantine-List-withIcon mantine-990sqx"
            >
              <div
                className="__mantine-ref-itemWrapper mantine-1hf9p21 mantine-List-itemWrapper"
              >
                <span
                  className="mantine-amxt3q mantine-List-itemIcon"
                >
                  <div
                    className="mantine-ThemeIcon-root mantine-17dkyrq"
                  >
                    <svg
                      className="icon icon-tabler icon-tabler-check"
                      fill="none"
                      height={12}
                      stroke="currentColor"
                      strokeLinecap="round"
                      strokeLinejoin="round"
                      strokeWidth={1.5}
                      viewBox="0 0 24 24"
                      width={12}
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                        stroke="none"
                      />
                      <path
                        d="M5 12l5 5l10 -10"
                      />
                    </svg>
                  </div>
                </span>
                <span>
                  <b>
                    Automatic Reading of Uploaded Prescriptions
                  </b>
                </span>
              </div>
            </li>
            <li
              className="mantine-List-item mantine-List-withIcon mantine-990sqx"
            >
              <div
                className="__mantine-ref-itemWrapper mantine-1hf9p21 mantine-List-itemWrapper"
              >
                <span
                  className="mantine-amxt3q mantine-List-itemIcon"
                >
                  <div
                    className="mantine-ThemeIcon-root mantine-17dkyrq"
                  >
                    <svg
                      className="icon icon-tabler icon-tabler-check"
                      fill="none"
                      height={12}
                      stroke="currentColor"
                      strokeLinecap="round"
                      strokeLinejoin="round"
                      strokeWidth={1.5}
                      viewBox="0 0 24 24"
                      width={12}
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                        stroke="none"
                      />
                      <path
                        d="M5 12l5 5l10 -10"
                      />
                    </svg>
                  </div>
                </span>
                <span>
                  <b>
                    Reminders to our Opted-In Customers 
                  </b>
                </span>
              </div>
            </li>
          </ul>
          <div
            className="mantine-Group-root mantine-inb06y"
          >
            <button
              className="mantine-UnstyledButton-root mantine-Button-root mantine-Group-child mantine-tlshck"
              data-button={true}
              onClick={[Function]}
              type="button"
            >
              <div
                className="mantine-3xbgk5 mantine-Button-inner"
              >
                <span
                  className="mantine-qo1k2 mantine-Button-label"
                  style={
                    Object {
                      "textTransform": undefined,
                    }
                  }
                >
                  Login
                </span>
              </div>
            </button>
            <button
              className="mantine-UnstyledButton-root mantine-Button-root mantine-Group-child mantine-11qo0g4"
              data-button={true}
              onClick={[Function]}
              type="button"
            >
              <div
                className="mantine-3xbgk5 mantine-Button-inner"
              >
                <span
                  className="mantine-qo1k2 mantine-Button-label"
                  style={
                    Object {
                      "textTransform": undefined,
                    }
                  }
                >
                  Register
                </span>
              </div>
            </button>
          </div>
        </div>
        <div
          className="mantine-Image-root animate__animated animate__backInRight mantine-15po0m8"
          style={
            Object {
              "height": 500,
              "width": "100%",
            }
          }
        >
          <figure
            className="mantine-11nhzn5 mantine-Image-figure"
          >
            <div
              className="mantine-qqmv3w mantine-Image-imageWrapper"
            >
              <img
                className="mantine-3y8yz3 mantine-Image-image"
                onError={[Function]}
                onLoad={[Function]}
                src="8000.jpg"
                style={
                  Object {
                    "height": 500,
                    "objectFit": "contain",
                    "width": "100%",
                  }
                }
              />
            </div>
          </figure>
        </div>
      </div>
    </div>
  </div>
</div>
`;
